source/fe/fe_system.cc

```diff
@@ -2302,6 +2303,29 @@ convert_generalized_support_point_values_to_dof_values (const std::vector<Vector
           continue;
         }
 
+      // If the base element is non-interpolatory, assign NaN to all DoFs
+      // associated to it.
+      if (!base_element.has_generalized_support_points())
+        {
+          base_dof_values.resize(n_base_dofs);
+
+          // To do this, we could really use a base_to_system_index()
+          // function, but that doesn't exist -- just do it by using the
+          // reverse table -- the amount of work done here is not worth
+          // trying to optimizing this.
+          for (unsigned int m = 0; m < multiplicity; ++m)
+            for (unsigned int i = 0; i < this->dofs_per_cell; ++i)
+              if (this->system_to_base_index(i).first ==
+                  std::make_pair(base, m))
+                dof_values[i] = std::numeric_limits<double>::quiet_NaN();
+
+          current_vector_component += multiplicity * n_base_components;
+          continue;
```

Comment:
I'd rather you use `else { ... }` for the remaining part of this block.

Or, even better, reverse the two cases and treat the situation that we *have* generalized support points first in the `if`, and the less likely case in the `else`.